### PR TITLE
Add link_to erb helper to render links to the artifact reference.

### PR DESCRIPTION
**Pull request by minuscorp**
~Test that actually works. (help appreciated)~

~This won't work for now as we cannot serve static files, I'll add it in a follow up PR but this is mergeable, to open the artifact just copy the link target in a new tab _et voilà_~

Not it works 😄 mergeable!! 🚀

### CODE REVIEW

**Comment by taquitos** [comment]
`This won't work for now`

Could you describe what in this PR does work?

File: app/features/build/build_controller.rb [hunk 1]
```diff
@@ -38,7 +38,7 @@ class BuildController < AuthenticatedControllerBase
         # if the server was restarted, we're gonna end here in this code block
         build_log_artifact = build.artifacts.find do |current_artifact|
           # We can improve the detection in the future, to actually mark an artifact as "default output"
-          current_artifact.type == "log" && current_artifact.reference.end_with?("fastlane.log")
+          current_artifact.type.include?("log") && current_artifact.reference.end_with?("fastlane.log")
```
Comment (KrauseFx) [inline app/features/build/build_controller.rb]: Why include?

File: app/features/build/views/build.erb [hunk 2]
```diff
@@ -58,7 +58,16 @@
       <% build.artifacts.each do |current_artifact| %>
         <tr>
           <td><%= current_artifact.type %></td>
-          <td><%= current_artifact.provider.retrieve!(artifact: current_artifact) %></td>
+          <td>
+            <% @reference = current_artifact.provider.retrieve!(artifact: current_artifact) %>
+            <% if File.exist?(@reference) %>
+              <a target="_blank" href="file:///<%= @reference %>">Download</a>
```
Comment (KrauseFx) [inline app/features/build/views/build.erb]: I tried this, and it didn't work on Chrome for me?

**Comment by taquitos** [comment]
Does something in this patch not work? I don't understand the comment. As in, if we merge this, are we merging something that is broken?

**Comment by minuscorp** [comment]
The only thing that does not work is that the browser does not open any `file:///` href. But if you copy the target link in a new tab your get the artifact.
But the thing that the browser does not allow to open that kind of files is for "security reasons"

File: app/features/build/build_controller.rb (hunk 1 above)
Comment (minuscorp) [reply app/features/build/build_controller.rb]: I'm taking more information that just "log" -> "verbose.log" or "log" now.

**Comment by minuscorp** [comment]
Fixed everything, ready to alpha!! 🚀🚀

File: Gemfile.lock [hunk 3]
```diff
@@ -1,14 +1,12 @@
 GIT
   remote: https://github.com/fastlane/TaskQueue
   revision: 5b3f459be5c451cc9ab25f6819d2eeb67be5e56d
-  ref: 5b3f459
```
Comment (KrauseFx) [inline Gemfile.lock]: Please don't commit Gemfile.lock changes as part of a PR

File: Gemfile.lock [hunk 4]
```diff
@@ -1,14 +1,12 @@
...
   specs:
     taskqueue (0.0.0)
 
 GIT
   remote: https://github.com/fastlane/fastfile-parser
-  revision: 122825dfb9370a6a2008e2e5ca8613dc6602b45d
-  ref: 122825d
+  revision: 4ee3b130e4d42bd61ee4134708a48e05c00d44b2
```
Comment (taquitos) [inline Gemfile.lock]: let's not commit this.

File: Gemfile.lock (hunk 3 above)
Comment (KrauseFx) [reply Gemfile.lock]: Makes it more work to test and merge 😢 

```
From https://github.com/minuscorp/ci
 * branch            link-artifacts -> FETCH_HEAD
error: Your local changes to the following files would be overwritten by merge:
	Gemfile.lock
Please commit your changes or stash them before you merge.
```

**Comment by KrauseFx** [comment]
Oh nice, the log download works great, but it breaks other artifacts, e.g. the `SCAN_DERIVED_DATA_PATH`

screen shot 2018-04-05 at 5 08 34 pm

screen shot 2018-04-05 at 5 08 39 pm

File: app/features/build/build_controller.rb [hunk 5]
```diff
@@ -19,6 +19,36 @@ class BuildController < AuthenticatedControllerBase
...
+    get "#{HOME}/*/artifact/*" do |project_id, build_number, artifact_id|
+      build_number = build_number.to_i
+
+      project = user_project_with_id(project_id: project_id)
+      build = project.builds.find { |b| b.number == build_number }
+
+      artifact = build.artifacts.find { |find_artifact| find_artifact.id == artifact_id }
+
+      if artifact.nil?
+        status(404) # Not foend
```
Comment (KrauseFx) [inline app/features/build/build_controller.rb]: spelling

**Comment by minuscorp** [comment]
Weird... He is checking the existence of the file in the filesystem and because it cannot find it, it assumes it is an URL, that obviously it is not, I'll try to fix it

**Comment by minuscorp** [comment]
This is mergeable btw!

**Pull request merged**